core/src/main/java/aima/core/search/basic/informed/AStarSearch.java

```diff
@@ -121,6 +121,10 @@ public AStarSearch(ToDoubleFunction<Node<A, S>> h) {
 		return searchController.solution(node);

```

Comment:
Yes, I have already taken this into account. Also for searchController.failure(), nodeFactory.newRootNode() and nodeFactory.newChildNode()